components/datetime/src/format/time_zone.rs

```diff
@@ -54,258 +54,6 @@ where
     T: TimeZoneInput,
     W: fmt::Write + ?Sized,
 {
-    match &time_zone_format.kind {
-        TimeZoneFormatKind::Pattern(patterns) => {
-            write_pattern(time_zone_format, time_zone, patterns, w)
-        }
-        TimeZoneFormatKind::Config(config) => write_config(time_zone_format, time_zone, *config, w),
-    }
-}
-
-fn write_config<T, W>(
-    time_zone_format: &TimeZoneFormat,
-    time_zone: &T,
-    config: TimeZoneFormatConfig,
-    w: &mut W,
-) -> Result<(), Error>
-where
-    T: TimeZoneInput,
-    W: fmt::Write + ?Sized,
-{
-    match config {
-        TimeZoneFormatConfig::GenericNonLocationLong => {
-            time_zone_format
-                .long_generic_non_location_format(w, time_zone)
-                .or_else(|_| time_zone_format.generic_location_format(w, time_zone))
-                .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?;
-        }
-        TimeZoneFormatConfig::GenericNonLocationShort => {
-            time_zone_format
-                .short_generic_non_location_format(w, time_zone)
-                .or_else(|_| time_zone_format.generic_location_format(w, time_zone))
-                .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?;
-        }
-        TimeZoneFormatConfig::GenericLocation => {
-            time_zone_format
-                .generic_location_format(w, time_zone)
-                .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?;
-        }
-        TimeZoneFormatConfig::SpecificNonLocationLong => {
-            time_zone_format
-                .long_specific_non_location_format(w, time_zone)
-                .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?;
-        }
-        TimeZoneFormatConfig::SpecificNonLocationShort => {
-            time_zone_format
-                .short_specific_non_location_format(w, time_zone)
-                .or_else(|_| time_zone_format.localized_gmt_format(w, time_zone))?;
-        }
-        TimeZoneFormatConfig::LocalizedGMT => {
-            time_zone_format.localized_gmt_format(w, time_zone)?;
-        }
-        TimeZoneFormatConfig::Iso8601(iso_format, iso_minutes, iso_seconds) => {
-            time_zone_format.iso8601_format(w, time_zone, iso_format, iso_minutes, iso_seconds)?;
-        }
-    }
-    Ok(())
-}
-
-fn write_pattern<T, W>(
-    time_zone_format: &TimeZoneFormat,
-    time_zone: &T,
-    patterns: &DataPayload<PatternPluralsFromPatternsV1Marker>,
-    w: &mut W,
-) -> Result<(), Error>
-where
-    T: TimeZoneInput,
-    W: fmt::Write + ?Sized,
-{
-    let pattern = patterns
-        .get()
-        .0
-        .clone()
-        .expect_pattern("Expected a single pattern");
-    for item in pattern.items.iter() {
-        match item {
-            PatternItem::Field(field) => write_field(field, time_zone_format, time_zone, w)?,
-            PatternItem::Literal(ch) => w.write_char(ch)?,
```

Comment:
do we still need `PatternItem::Literal(ch) => w.write_char(ch)`?

Reply:
I think the expectation here would be that this analyzes the pattern and calls `.load_data...()` to build the correct object format object instead.

Reply:
reverted the change and we still need `PatternItem::Literal(ch) => w.write_char(ch)`